ssvsigner/client.go

```
@@ -0,0 +1,224 @@
+package ssvsigner
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/carlmjohnson/requests"
+	"github.com/ssvlabs/ssv/logging/fields"
+	"go.uber.org/zap"
+
+	"github.com/ssvlabs/ssv/ssvsigner/web3signer"
+)
+
+type Client struct {
+	logger     *zap.Logger
+	baseURL    string
+	httpClient *http.Client
+}
+
+func NewClient(baseURL string, opts ...ClientOption) *Client {
+	baseURL = strings.TrimRight(baseURL, "/")
```

I am not sure if `baseURL` (which comes from the `SSVSignerEndpoint` config) is validated at any point.  
It might be worth validating whether it is a valid URL and failing fast if it’s not.  
Maybe we already have a utility function within the codebase for this purpose?

---

It's not, but the client requests operator identity right after connection, so it would fail. But yes, perhaps validating the URL would make it more user-friendly, I'll add it